### PR TITLE
Fixed issue where taint sources werent initialized and propagated

### PR DESCRIPTION
**Pull request by thorulf4**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -210,17 +210,19 @@ void benchmark_all_multi_taint(antlr4::ANTLRInputStream& prog, std::string file_
...
+        program = parse_to_cfg_transformer(prog, call_counts);
+        Stopwatch gpu_watch;
+        auto gpu_nodes = multi_bit_cuda_worklist_analysis(program);
+        gpu_watch.save_time<Microseconds>();
+
+        if(!state_multi_equality(comparable_cpu_nodes, gpu_nodes)){
+            std::cout << "###### cpu != gpu #####" << std::endl;
+        }
+        Stopwatch::add_line();
     }
```
Comment (NicklasHugoy) [inline src/main.cpp:225]: Why the scope?

File: src/multi_taint_analysis.cpp [hunk 2]
```diff
@@ -18,14 +18,22 @@ BitVector join(Node& node, std::map<Node*, std::vector<BitVector>*>& states, int
...
 
-        (*states[&node])[source_index] |= joined_state & transfer->join_mask;
+        if(joined_state.bitfield == 0)
+            continue;
+
+        int iterations = 0;
+
+        state[source_index] |= joined_state & transfer->join_mask;
         do{
+            iterations++;
```
Comment (NicklasHugoy) [inline src/multi_taint_analysis.cpp:34]: What is `iterations` used for?

File: src/multi_taint_analysis.cpp [hunk 3]
```diff
@@ -55,8 +66,10 @@ void cpu_multi::worklist(std::vector<StatefulNode<std::vector<BitVector>>>& node
         StatefulNode<std::vector<BitVector>>& currentNode = nodes[index];
 
         std::vector<BitVector> oldState = currentNode.get_state();
-
-        analyze(*currentNode.node, *currentNode.states, transfers.cbegin() + node_to_start_transfer[index], source_count);
+        int offset = node_to_start_transfer[index];
+        auto begin = transfers.cbegin();
```
Comment (NicklasHugoy) [inline src/multi_taint_analysis.cpp:70]: `begin` is unused

**Review by ZcortezZ — APPROVED** [review]

File: src/main.cpp (hunk 1 above)
Comment (thorulf4) [reply src/main.cpp:225]: Was used to delete old data, but i have removed it

File: src/multi_taint_analysis.cpp (hunk 2 above)
Comment (thorulf4) [reply src/multi_taint_analysis.cpp]: debug variable; removed

**Review by NicklasHugoy — APPROVED** [review]

**Review by ZcortezZ — APPROVED** [review]

**Pull request merged**